doc/source/data/key-concepts.rst

```diff
@@ -33,153 +26,93 @@ Since a Dataset is just a list of Ray object references, it can be freely passed
+You can also change just the number of blocks of a Dataset using :meth:`ds.repartition() <ray.data.Dataset.repartition>`. Repartition has two modes, ``shuffle=False``, which performs the minimal data movement needed to equalize block sizes, and ``shuffle=True``, which performs a full distributed shuffle:
 
 .. image:: images/dataset-shuffle.svg
-   :width: 650px
    :align: center
 
 ..
   https://docs.google.com/drawings/d/132jhE3KXZsf29ho1yUdPrCHB9uheHBWHJhDQMXqIVPA/edit
-  
+
+Datasets shuffle can scale to processing hundreds of terabytes of data. See the :ref:`Performance Tips Guide <shuffle_performance_tips>` for an in-depth guide on shuffle performance.
+
 Fault tolerance
 ===============
 
-Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, a ``Dataset`` will be automatically recovered by Ray in case of failures. This works through **lineage reconstruction**: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate them by re-executing the task(s) that created them.
+Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, a Dataset will be automatically recovered by Ray in case of failures. This works through *lineage reconstruction*: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate them by re-executing the tasks that created them.
 
 There are a few cases that are not currently supported:
-1. If the original creator of the ``Dataset`` dies. This is because the creator stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the ``Dataset``.
-2. For a :meth:`DatasetPipeline.split() <ray.data.DatasetPipeline.split>`, we do not support recovery for a consumer failure. When there are multiple consumers, they must all read the split pipeline in lockstep. To recover from this case, the pipeline and all consumers must be restarted together.
-3. The ``compute=actors`` option for transformations.
 
-Execution and Memory Management
-===============================
+ * If the original worker process that created the Dataset dies. This is because the creator stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the Dataset.
+ * When ``compute=ActorPoolStrategy()`` is specified for transformations.
 
-See :ref:`Execution and Memory Management <data_advanced>` for more details about how Datasets manages memory and optimizations such as lazy vs eager execution.
+-----------------
+Dataset Pipelines
+-----------------
 
--------------------------
-Resource Allocation Model
```

Comment:
Shall we keep this and move it to "Memory Management" page? IMO the inter-op of Ray libraries is not just APIs, but also execution model (in particular the resource allocation/isolation).

Reply:
Nack. I don't think this level of detail is appropriate / necessary. At the AIR level, users shouldn't know that placement groups exist.

Reply:
Just to be clear, I was meaning the Memory Management page under User Guide (https://docs.ray.io/en/master/data/memory-management.html). It makes sense to remove from Key Concepts for easier onramp, but do you think there is also no need to talk about such mechanisms after user onboarding or for those want to learn more?

Reply:
I see the question. I think overall I'd prefer to get rid of the special case behavior of Datasets when scheduling tasks with placement groups. If we default placement groups to not applying to tasks, then there is no longer a need to document this behavior as much.

Reply:
Update: actually it seems like a huge pain to change the default, and won't work anyways for ActorPoolStrategy. Instead, I decided to add a section to performance tips on the placement group behavior:

```
+Datasets and Placement Groups
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Datasets configures its tasks and actors to use the cluster-default
+scheduling strategy ("DEFAULT"). You can inspect this configuration variable here:
+``ray.data.context.DatasetContext.get_current().scheduling_strategy``. This
+scheduling strategy will schedule these tasks and actors outside any present
+placement group. If you want to force Datasets to schedule tasks within the
+current placement group, you can set
+``ray.data.context.DatasetContext.get_current().scheduling_strategy = None``.
```

Reply:
Sounds good.